### ISSUE TITLE
[Suggestion] Auto-place Torches

### ISSUE THREAD

**Issue by Lolfaceftw** [cabaletta/baritone#971]
Good day people! I would like to suggest an auto-place function for torches so that while mining, you won't worry mobs spawning behind you or previous tunnels you built :)

#autotorch (on/off auto torch)
#autotorchradius (specifies what light level will the torch be placed on)

Adding this suggestion will make less worries for mobs while afk'ing. Although #avoidance has this, adding light while mining is better. It also guides your way.

**Comment by Lolfaceftw** [comment]
up

**Comment by usereewrjrjeejr** [comment]
Turn on kill aura

**Comment by Lolfaceftw** [comment]
Yes, for Impact. But what if I wanted to use 1.14? Then I'd have to pay for Impact yet not many people will choose to buy a hack client. I'd use the stand-alone Baritone due to its simplicity.

**Comment by jonnyawsom3** [comment]
^ or people like me that just use baritone standalone

**Comment by BlueBlazeTech** [comment]
its also nice to have some light, i guess?

**Comment by jonnyawsom3** [comment]
personally i vote that if this gets implemented, that the toggle command should be "lightmeup"

**Comment by Lolfaceftw** [comment]
That's a good suggestion. I hope that this feature will be added on future releases.

**Comment by jonnyawsom3** [comment]
considering we're already at 1.2.9, i guess we'll just have to slap another decimal or two onto it

**Comment by Xephorix** [comment]
@Lolfaceftw agreed.

**Comment by Lolfaceftw** [comment]
@Xephorix Do you have any proposition to improve this suggestion?

**Comment by HyCore** [comment]
So... Autotorch has been implemented or not...

**Comment by dominikaaaa** [comment]
look at the issue status

**Comment by dominikaaaa** [comment]
# ♿♿♿

**Comment by HyCore** [comment]
ho yeah sorry, i'm pretty bad on baritone, i'm learning how all of that code work, and i just discovered i wasn't using the good version. So, thank you ^^

**Comment by Crayder** [comment]
So like, what's the issue here? Does anyone have anything AGAINST this being added?

**Comment by Crayder** [comment]
At least make baritone NOT break the torches (by default) that are already on a path. Why does it have to break torches

**Comment by dominikaaaa** [comment]
Nobody is against this being added. I don't see where you got that idea from. As for the last part, there's already a bug report opened for that

**Comment by arthurstdenis** [comment]
Meteor client 1.18.1 (baritone) world > Spawn proofer

**Comment by PlasmaHH** [comment]
Just came here to make the very same suggestion. As the most easy implementation I could imagine every time on a path a block under a light threshold (probably 0 default) is encountered a torch will be placed on it. There are tons of edge cases where this might not lead to the desired result, but its a 99% solution that will make many people happy and I think it might not be too hard to implement the next time this code is touched...

**Comment by TomieAi** [comment]
hmm tunnel will be awesome using this.

**Comment by Bredosen** [comment]
Could be a great implementation.

**Comment by TomieAi** [comment]
bruh its been 2019 hahaha xD I guess GTA 6 will come first before this.

**Comment by pizzav-xyz** [comment]
added?

**Comment by ycMia** [comment]
That's really a feature come after GTA6, nevermind, I hope some day the community would have it done.

**Comment by ingenarel** [comment]
+1

**Comment by TomieAi** [comment]
> bruh its been 2019 hahaha xD I guess GTA 6 will come first before this.

ngl my prediction is coming true soon hehe.. they implement parkour much complex than placing torch lol.. i think they can do it .. trust

**Comment by kekkykekkek** [comment]
Implemented in [cabaletta/baritone#5118](https://github.com/cabaletta/baritone/pull/5118) TorchProcess places torches on nearby dark blocks, off by default behind autoTorch
